### PR TITLE
Updated protobuf and jsonschema.

### LINKED ISSUE #2814: Feature request: Update protobuf & jsonschema dependencies
### Prerequisites

- [X] I checked the [documentation](https://docs.fetch.ai/aea/) and made sure this feature does not already exist.
- [X] I checked the [existing issues](https://github.com/fetchai/agents-aea/issues) to make sure this feature has not already been requested.
- [X] I have read the [code of conduct](https://github.com/fetchai/agents-aea/blob/main/CODE_OF_CONDUCT.md) before creating this issue.

### Problem

Importing `aea` version `1.2.3`, some packages are not compatible with `eth-brownie` module such as `jsonschema` & `protobuf`.
The output snippets after the installation of the deps:
```console
[pipenv.exceptions.InstallError]: The conflict is caused by:
[pipenv.exceptions.InstallError]:     aea 1.2.3 depends on jsonschema<5.0.0 and >=4.0.0
[pipenv.exceptions.InstallError]:     eth-brownie 1.19.1 depends on jsonschema==3.2.0
[pipenv.exceptions.InstallError]: 

[pipenv.exceptions.InstallError]: The conflict is caused by:
[pipenv.exceptions.InstallError]:     aea 1.2.3 depends on protobuf<=3.19.4 and >=3.19.0
[pipenv.exceptions.InstallError]:     eth-brownie 1.19.1 depends on protobuf==3.20.1
[pipenv.exceptions.InstallError]: 
```

### Feature / Solution

Extend the range of versions of the dependencies in order to be compatible with brownie module as well. Such as:
```console
jsonschema = "^3.2.0"
protobuf = ">=3.19.4,<4.0.0"
```

### Alternatives

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by mariacarmina**
## Proposed changes

Updated `protobuf` and `jsonschema` versions for further development.

## Fixes

Fixes #2814

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](https://github.com/fetchai/agents-aea/blob/main/CONTRIBUTING.md) doc
- [X] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [] Lint and unit tests pass locally with my changes and CI passes too
- [] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that code coverage does not decrease.
- [X] I have added necessary documentation (if appropriate)
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mariacarmina** [comment]
@solarw I saw that you opened a PR which is related with what I described in #2814. I have done some new changes on this branch and I was wondering if you can approve the CI workflows to run please. Thanks!

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -36,9 +36,12 @@ jobs:
         with:
           python-version: '${{ matrix.python-version }}'
       - name: Install dependencies
-        run: pip install tox==3.25.1
+        run: |
+          pip install tox==3.25.1
+          pip install poetry
       - name: 'Run check ${{ matrix.tox-job }}'
         run: |
+          poetry update --lock
```
Comment (solarw) [inline .github/workflows/workflow.yml:44]: please, dont do lock in CI, it's a long time operation
and actually should be done in repo and commited

File: pyproject.toml [hunk 2]
```diff
@@ -65,7 +65,7 @@ optional = true
 tox = "^3.26"
 py-sr25519-bindings = "^0.1.5"
 pylint = "==2.15.5"
-cosmpy = ">=0.6.0,<0.7.0"
+cosmpy = ">=0.5.1,<0.7.0"
```
Comment (solarw) [inline pyproject.toml:68]: the new release of cosmpy with dependencies improved is in it's way.

so in a while can be updated to cosmpy>=0.6.1,<0.7

**Review by solarw — COMMENTED** [review]
LGTM, couple of small comments and waiting till 0.6.1

File: pyproject.toml (hunk 2 above)
Comment (5A11) [reply pyproject.toml:68]: This is already done now (CosmPy 0.6.1 is out and addresses the jsonschema deps). 
Please revert this change.

**Comment by 5A11** [comment]
@mariacarmina Thanks for your PR. Please address the above comments, otherwise should be ready to go.

**Review by 5A11 — CHANGES_REQUESTED** [review]
The PR should be based off of and against `develop`and not `main`.

**Comment by 5A11** [comment]
@mariacarmina Thanks for your PR. Please address the above comments, otherwise should be ready to go.

File: pyproject.toml (hunk 2 above)
Comment (mariacarmina) [reply pyproject.toml:68]: At some point I had trouble locally to install cosmpy 0.6.0, that's why I have reverted to a previous version.

**Comment by mariacarmina** [comment]
> The PR should be based off of and against `develop`and not `main`.

I have changed the base to `develop`.

File: pyproject.toml [hunk 3]
```diff
@@ -37,10 +37,10 @@ aea = 'aea.cli:cli'
 [tool.poetry.dependencies]
 python = ">=3.8,<3.11"
 base58 = ">=1.0.3,<3.0.0"
-jsonschema = "^4.0.0"
+jsonschema = "^3.2.0"
```
Comment (5A11) [inline pyproject.toml:40]:
```suggestion
jsonschema = ">=3.2.0,<5"
```

**Comment by mariacarmina** [comment]
I have updated the `poetry.lock` file. The CI was complaining about `python-mbedtls (==2.2.0)` package which was present before.

**Comment by mariacarmina** [comment]
The remaining failing tests are: `python code quality checks` score which is 9.99/10 and `golang checks` on `MacOS` system.

**Comment by mariacarmina** [comment]
> @mariacarmina thanks for your contribution. CI is failing on some pylint issue. In the interest of time, we have created a PR #2815 that passes CI (mostly except flaky tests) and achieves the same. Do you want to have a look and if you are happy approve?

yes, I can take a look over that PR, sure. Thanks for letting me know.

**Comment by 5A11** [comment]
Thanks again for your contribution @mariacarmina and reviewing #2815. That PR is now merged and so this can be closed.

**Pull request closed without merging**